shared/teams/channel/tabs/index.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react'
+import * as Types from '../../../constants/types/teams'
+import * as ChatTypes from '../../../constants/types/chat2'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import flags from '../../../util/feature-flags'
+import capitalize from 'lodash/capitalize'
+
+export type TabKey = 'members' | 'attachments' | 'bots' | 'settings' | 'loading'
+
+export type OwnProps = {
+  teamID: Types.TeamID
+  conversationIDKey: ChatTypes.ConversationIDKey
+  selectedTab: TabKey
+  setSelectedTab: (t: TabKey) => void
+}
+
+export type Props = OwnProps & {
+  admin: boolean
+  error?: string
+  loadBots: () => void
+  loading: boolean
+}
+
+const TabText = ({selected, text}: {selected: boolean; text: string}) => (
+  <Kb.Text type="BodySmallSemibold" style={selected ? styles.tabTextSelected : styles.tabText}>
+    {text}
+  </Kb.Text>
+)
+
+type TabProps = {
+  key: TabKey
+  selectedTab: TabKey
+}
+
+const Tab = ({key, selectedTab}: TabProps) => (
```

Comment:
`key` is reserved and can't be used as a prop name

Reply:
That's the property I'm using...I did it this way to make the typing work as well as possible.

Reply:
If I rename the prop to, say, `name`, then react yells at me below for not having a `key` prop in an array.

Reply:
From my perspective I'm just telling typescript that the key prop that we already need is typed `TabKey`.

Reply:
React uses `key` to manage elements in a list - it's filtered out of props and can't be read from within the component. This is why the tab texts are empty.

Reply:
So I have to pass in the same thing to both a non-`key` prop and a `key` prop?

Reply:
Yes